packages/doenetml/src/Viewer/renderers/point.tsx

```diff
@@ -267,106 +274,116 @@ export default React.memo(function Point(props) {
+                }
 
-            //Protect against very small unintended drags
-            if (
-                !viaPointer ||
-                Math.abs(e.x - pointerAtDown.current[0]) > 0.1 ||
-                Math.abs(e.y - pointerAtDown.current[1]) > 0.1
-            ) {
-                dragged.current = true;
-            }
+                let [xmin, ymax, xmax, ymin] = board.getBoundingBox();
 
-            let [xmin, ymax, xmax, ymin] = board.getBoundingBox();
+                let xminAdjusted = xmin;
+                let xmaxAdjusted = xmax;
+                let yminAdjusted = ymin;
+                let ymaxAdjusted = ymax;
 
-            let xminAdjusted = xmin;
-            let xmaxAdjusted = xmax;
-            let yminAdjusted = ymin;
-            let ymaxAdjusted = ymax;
+                if (xmax < xmin) {
+                    [xmaxAdjusted, xminAdjusted] = [xminAdjusted, xmaxAdjusted];
+                }
+                if (ymax < ymin) {
+                    [ymaxAdjusted, yminAdjusted] = [yminAdjusted, ymaxAdjusted];
+                }
 
-            if (xmax < xmin) {
-                [xmaxAdjusted, xminAdjusted] = [xminAdjusted, xmaxAdjusted];
-            }
-            if (ymax < ymin) {
-                [ymaxAdjusted, yminAdjusted] = [yminAdjusted, ymaxAdjusted];
-            }
+                let xscale = xmaxAdjusted - xminAdjusted;
+                let yscale = ymaxAdjusted - yminAdjusted;
+
+                xmaxAdjusted -= xscale * 0.01;
+                xminAdjusted += xscale * 0.01;
+                ymaxAdjusted -= yscale * 0.01;
+                yminAdjusted += yscale * 0.01;
+
+                if (
+                    viaPointer &&
+                    pointAtDown.current &&
+                    pointerAtDown.current
+                ) {
+                    // the reason we calculate point position with this algorithm,
+                    // rather than using .X() and .Y() directly
+                    // is that attributes .X() and .Y() are affected by the
+                    // .setCoordinates function called in update().
+                    // Due to this dependence, the location of .X() and .Y()
+                    // can be affected by constraints of objects that the points depends on,
+                    // leading to a different location on up than on drag
+                    // (as dragging uses the mouse location)
+                    // TODO: find an example where need this this additional complexity
+                    var o = board.origin.scrCoords;
+                    calculatedX.current =
+                        (pointAtDown.current[1] +
+                            e.x -
+                            pointerAtDown.current[0] -
+                            o[1]) /
+                        board.unitX;
+                    calculatedY.current =
+                        (o[2] -
+                            // TODO: Is this correct? What is the `[2]` doing?
```

Comment:
Yes, it's correct, as mentioned in previous comments about homogeneous coordinates.

Reply:
There seems to be a lot of duplicated JSXGraph code. Would it be easy to pull out or is it all customized per component in subtle ways?

Reply:
For components represented by a single location, I expect it is all copy and paste. There are presumably customizations for objects, lines and more, based on multiple points.

You will find a ton of copy and paste, though, in this renderer code.